From f2742fa411bc1aacbc57d0417c2a4ff0a35e1736 Mon Sep 17 00:00:00 2001
From: "Elias S. G. Carotti" <eliascrt@amazon.it>
Date: Thu, 17 Sep 2026 11:28:39 +0200
Subject: [PATCH] Include intrin.h before using _sub_overflow_i32 on MSVC

JXL_COMPILER_MSVC is defined by compiler_specific.h, so guarding the intrin.h
include with it before that header is included leaves the include out. MSVC
then fails with C3861 on _sub_overflow_i32 in the translation units that do not
pull in intrin.h through some other header.
---
 lib/jxl/base/common.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/lib/jxl/base/common.h b/lib/jxl/base/common.h
index 53d0959..12d9458 100644
--- a/lib/jxl/base/common.h
+++ b/lib/jxl/base/common.h
@@ -18,12 +18,12 @@
 #include <type_traits>
 #include <vector>
 
+#include "lib/jxl/base/compiler_specific.h"
+
 #if JXL_COMPILER_MSVC
 #include <intrin.h>
 #endif
 
-#include "lib/jxl/base/compiler_specific.h"
-
 namespace jxl {
 // Some enums and typedefs used by more than one header file.
 
-- 
2.43.0

